test/functional/rpc_whitelistdefault.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+"""
+A test for rpcwhitelistdefault states.
+"""
+
+from test_framework.test_framework import BitcoinTestFramework
+import os
+from test_framework.util import (
+    assert_equal,
+    get_datadir_path,
+    str_to_b64str
+)
+import http.client
+import urllib.parse
+
+def rpccall(node, user, password, method):
+    url = urllib.parse.urlparse(node.url)
+    headers = {"Authorization": "Basic " + str_to_b64str('{}:{}'.format(user, password))}
+    conn = http.client.HTTPConnection(url.hostname, url.port)
+    conn.connect()
+    conn.request('POST', '/', '{{"method": "{}"}}'.format(method), headers)
+    resp = conn.getresponse()
+    conn.close()
+    return resp.code
+
+class RPCWhitelistDefaultTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 4
```

Comment:
perhaps add: `self.setup_clean_chain = False` for clarity

Reply:
It doesn't matter anyway

Reply:
It is false by default (see test_framework.py::L96), but most tests set it explicitly even when false. It matters a bit for test speed as it controls whether or not to use the cached data directories for the test setup.